skglm/solvers/group_bcd_solver.py

```diff
@@ -119,3 +139,15 @@ def _bcd_epoch(X, y, w, Xw, datafit, penalty, ws):
             if old_w_g[idx] != w[j]:
                 Xw += (w[j] - old_w_g[idx]) * X[:, j]
     return
+
+
+def _construct_grad(X, y, w, Xw, datafit: QuadraticGroup, ws):
+    """Compute the -gradient according to each group in ``ws``.
+
+    Note: -gradients are stacked in a 1d array (e.g. [-grad_g1, -grad_g2, ...]).
+    """
+    grads = np.array([])
```

Comment:
here, preallocate an array of size n_features and fill it

Reply:
- ``_compute_grad`` act on a working set rather than all groups, hence it's not relevant to init grad with ``n_features``
- groups have not necessarily equal sizes hence computing the actual size of ``grad`` requires additional computation

Reply:
@Badr-MOUFAD can't you do something like this: `grads = np.zeros(len(ws))`?

Reply:
Could you do grads = np.zeros([n_features_in_ws]) ? and a simple computation for `n_features_in_ws` ?